history.go

```diff
@@ -7,37 +7,60 @@ import (
 )
 
 var (
-	ErrBadEpoch    = errors.New("[merkletree] Next epoch's STR has bad epoch")
-	ErrSTRNotFound = errors.New("[merkletree] STR not found")
+	ErrBadEpoch           = errors.New("[merkletree] Next epoch's STR has bad epoch")
+	ErrSTRNotFound        = errors.New("[merkletree] STR not found")
+	ErrMalformedHashChain = errors.New("[merkletree] Malformed hash chain")
 )
 
+type Epoch int64
+
 type History struct {
-	str           *SignedTreeRoot
-	epochInterval int64
+	currentSTR    *SignedTreeRoot
+	hashChain     map[Epoch]*SignedTreeRoot
+	epochs        []Epoch // slice of epochs in hashChain
+	epochInterval Epoch
```

Comment:
Yes. They has the same value. But I think it is quicker to have this field in the `History` struct, instead of referencing to the value stored in the STR's policy (which is actually the value stored in `Policies` struct in `MerkleTree`)